Strata/DDM/HNF.lean

```diff
@@ -0,0 +1,40 @@
+/-
+  Copyright Strata Contributors
+
+  SPDX-License-Identifier: Apache-2.0 OR MIT
+-/
+
+module
+public import Strata.DDM.AST
+import Strata.DDM.Util.Array
+
+public section
+namespace Strata
+
+/--
+Array ofelements whose sizes are bounded by a value.
+-/
+public abbrev SizeBounded (α : Type _) [SizeOf α] {β} [SizeOf β] (e : β) (c : Int) := { a : α // sizeOf a ≤ sizeOf e + c }
+
+namespace ExprF
+
+/--
+Head-normal form for an expression consists of an operation
+-/
+structure HNF {α} (e : ExprF α) where
```

Comment:
Is this meant to be used in ASTs?

Reply:
This is used by the elaborator to normalize expressions.